components/feral-connectd/cdp.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/gorilla/websocket"
+)
+
+// CDP Methods
+const (
+	NavigateMethod = "Page.navigate"
+	EvaluateMethod = "Runtime.evaluate"
+)
+
+type CDPClient struct {
+	conn   *websocket.Conn
+	reqID  int
+	config *Config
+	mu     sync.Mutex
+	done   chan struct{}
```

Comment:
This chan seems to be redundant. If we just want to check if the conn is closed, we could just use a bool to be as simple.